cdap-api/src/main/java/co/cask/cdap/api/lineage/operation/Join.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.operation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents join operation.
+ */
+public class Join extends Operation {
+  private final Constraint constraint;
+  private final Type joinType;
+  private final List<String> outputs;
+
+  /**
+   * Create an instance of join operation.
+   * @param name the name of the join operation
+   * @param description the description of the join operation
+   * @param constraint the constraint for the join operation
```

Comment:
I think this kind of Javadoc is kind of useless :)  
Either describe something beyond the name of the parameter, or omit this.

Reply:
added link to the class javadoc.